addon-test-support/setup-middleware-reporter.ts

```diff
@@ -149,15 +159,28 @@ export function setupMiddlewareReporter() {
 
   QUnit.testDone(pushTestResult);
 
-  QUnit.done(async function () {
-    let response = await fetch('/report-violations', {
-      method: 'POST',
-      headers: {
-        'Content-Type': 'application/json',
-      },
-      body: JSON.stringify(TEST_SUITE_RESULTS),
+  if (window.Testem) {
+    window.Testem.afterTests(async function (_config, _data, callback) {
+      try {
+        await sendViolationsToServer();
+      } finally {
+        callback();
+      }
+    });
+  } else {
+    QUnit.done(async function () {
```

Comment:
falling back to QUnit.done in the rare case that folks are running in an environment other than Testem. I remembered some teams save their runners on S3 or other file stores for debugging purposes.

Reply:
The fallback here is appreciated as a review and consumer!